roles/dcos_bootstrap/tasks/main.yml

```diff
@@ -280,13 +280,38 @@
     creates: "{{ download_path }}/genconf/serve/windows/upgrade_from_{{ item }}/"
   when:
     - "dcos['version'] is version('1.9', '>=')"
-    - "dcos['download_win'] is defined"
+    - dcos['download_win'] is defined
   loop: >
     {{ groups['dcos'] | default([]) | map('extract',hostvars,'ansible_local')| list |
      select('defined') | selectattr('dcos_installation', 'defined') |
      map(attribute='dcos_installation') | map(attribute='version') | list |
      union([dcos['version_to_upgrade_from'] | default(dcos['version']) ]) | unique }}
 
+- name: Get upgrade directory hash
+  shell: |
+    set -o pipefail
+    ls -td -- */ | head -n 1 | cut -d'/' -f1
+  args:
+    chdir: "{{ download_path }}/genconf/serve/windows/upgrade_from_{{ item }}"
+  changed_when: false
+  register: upgrade_win_dir_hash
+  when: "dcos['version'] is version('1.9', '>=')"
+  loop: >
+    {{ groups['dcos'] | default([]) | map('extract',hostvars,'ansible_local')| list |
+     select('defined') | selectattr('dcos_installation', 'defined') |
+     map(attribute='dcos_installation') | map(attribute='version') | list |
+     union([dcos['version_to_upgrade_from'] | default(dcos['version']) ]) | unique }}
+   
+- name: Create latest/ symlink for upgrade_win.ps1
+  file:
+    path: "{{ download_path }}/genconf/serve/windows/upgrade_from_{{ item.item }}/latest"
+    src:  "{{ item.stdout }}"
+    state: link
+  when: "dcos['version'] is version('1.9', '>=')"
+  loop: "{{ upgrade_win_dir_hash.results }}"
+   
```

Comment:
the whole windows stuff should be in a conditional:

at least two conditions must be given win url and enable windows in dcos conf